Source/SvgDocument.cs

```diff
@@ -175,6 +175,68 @@ public virtual SvgElement GetElementById(string id)
             return (this.GetElementById(id) as TSvgElement);
         }
 
+        /// <summary>
+        /// Validate whether the system has GDI+ capabilities (non Windows related).
+        /// </summary>
+        /// <returns>Boolean whether the system is capable of using GDI+</returns>
+        public static bool SystemIsGdiPlusCapable()
+        {
+            try 
+            {
+                EnsureSystemIsGdiPlusCapable();
+            }
+            catch(SvgGdiPlusCannotBeLoadedException)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Ensure that the running system is GDI capable, if not this will yield a
+        /// SvgGdiPlusCannotBeLoadedException exception.
+        /// </summary>
+        public static void EnsureSystemIsGdiPlusCapable()
+        {
+            try
+            {
+                var matrix = new Matrix(0, 0, 0, 0, 0, 0);
+            }
+            // GDI+ loading errors will result in TypeInitializationExceptions, 
+            // for readability we will catch and wrap the error
+            catch (Exception e)
+            {
+                if (ExceptionCaughtIsGdiPlusRelated(e))
+                {
+                    // Throw only the customized exception if we are sure GDI+ is causing the problem
+                    throw new SvgGdiPlusCannotBeLoadedException(e);
+                }   
+                //Ignore any other form of error

```

Comment:
I would rather re-raise here, as there must be something really wrong if this raises.

Reply:
Fair point, I would not expect it to happen, but I share the vision that this should yield an error for the caller to handle.